src/eth/provider.ts

```diff
@@ -0,0 +1,327 @@
+// import dayjs from 'dayjs'
+
+import { allSettled } from 'utils/allSettled'
+import { Collectible, CollectibleState } from 'utils/types'
+
+import {
+  assetToCollectible,
+  getAssetIdentifier,
+  isAssetValid
+  // isNotFromNullAddress,
+  // transferEventToCollectible
+} from './helpers'
+import { OpenSeaClient } from './opensea'
+import {
+  OpenSeaCollection,
+  // OpenSeaEvent,
+  // OpenSeaEventExtended,
+  OpenSeaNft,
+  OpenSeaNftExtended,
+  OpenSeaNftMetadata
+} from './types'
+
+export class EthereumCollectiblesProvider {
+  private readonly openSeaClient: OpenSeaClient
+
+  constructor(openSeaClient: OpenSeaClient) {
+    this.openSeaClient = openSeaClient
+  }
+
+  private async getNftsForMultipleWallets(wallets: string[]): Promise<OpenSeaNft[]> {
+    return allSettled(
+      wallets.map((wallet) => this.openSeaClient.getNftsForWallet(wallet))
+    ).then((results: PromiseSettledResult<OpenSeaNft[]>[]) =>
+      results
+        .map((result, i) => ({ result, wallet: wallets[i] }))
+        .filter(({ result }) => result.status === 'fulfilled')
+        .map(
+          ({ result, wallet }) =>
+            (result as PromiseFulfilledResult<OpenSeaNft[]>).value?.map(
+              (nft) => ({ ...nft, wallet })
+            ) || []
+        )
+        .flat()
+    )
+  }
+
+  // private async getNftTransferEventsForMultipleWallets(
```

Comment:
comment

Reply:
on purpose lol. at least for now. this calls allows us to get a few more nfts, but the perf implication is heavy. so we decided to go with this for now. maybe we also simply remove the comment? no strong opinions there

Reply:
you know what. let's remove the comments. we have git history for reference, in case we want to add it back